### PR TITLE
Adding unit tests for components

### PR DESCRIPTION
**Pull request by renjithgr**
- Run `yarn test` to run the unit tests.
- Unit test step is added to the CI pipeline

### CODE REVIEW

File: src/components/code-editor/code-editor.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        initialValue={"initial value"}
+        readOnly={false}
+        dataTestId={"test-id"}
+      />
+    );
+
+    expect(screen.getByTestId("test-id")).not.toHaveAttribute("readOnly");
+  });
+
+  test("should call the callback when input changes", async () => {
```
Comment (jamesgeorge007) [inline src/components/code-editor/code-editor.test.js:48]:
```suggestion
  test("should invoke the callback when input changes", async () => {
```

File: src/components/select-box/selectBox.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { Context } from "../../App";
+import SelectBox from "./SelectBox";
+
+test("should call the callback when value changes", async () => {
```
Comment (jamesgeorge007) [inline src/components/select-box/selectBox.test.js:8]:
```suggestion
test("should invoke the callback when value changes", async () => {
```

File: src/components/transform/transform.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  expect(firstCodeEditor).toHaveAttribute(
+    "data-placeholder",
+    contextValue.inputCss
+  );
+  expect(firstCodeEditor).toHaveAttribute("data-initial-value", "");
+  expect(firstCodeEditor).toHaveAttribute("data-type", "input");
+  expect(firstCodeEditor).not.toHaveAttribute("readOnly");
+
+  const secondCodeEditor = getAllByTestId("code-editor")[1];
```
Comment (jamesgeorge007) [inline src/components/transform/transform.test.js:42]:
```suggestion
  const [firstCodeEditor, secondCodeEditor] = getAllByTestId("code-editor");

  expect(firstCodeEditor).toHaveAttribute(
    "data-placeholder",
    contextValue.inputCss
  );
  expect(firstCodeEditor).toHaveAttribute("data-initial-value", "");
  expect(firstCodeEditor).toHaveAttribute("data-type", "input");
  expect(firstCodeEditor).not.toHaveAttribute("readOnly");

```

**Comment by renjithgr** [comment]
Your suggestions are applied.

**Pull request merged**